payment-account-age-witness.adoc

```diff
@@ -0,0 +1,196 @@
+
+_Side note:
+Each offer contains the ID of one specific payment account (we don't support multiple accounts for the same payment method)._
+
+=== Verification
+
+When a trader takes an offer both users are exchanging in the trade process a signed nonce, the pubKey and the salt for the hash used in the AgeWitness. With that data the other peer can verify that the other trader is the owner of the AgeWitness data and that the hash is matching the account data used for the trade. The date can be verified as well and is used to set the permitted trade limit. Any violation of those rules would lead to a failed trade (before the deposit transaction gets created - trading fees would be lost).
+
+As the date of both users will differ at least sightly we use a tolerance window to avoid problems with edge cases. E.g. The maker send the taker his local date and his AgeWitness. The taker can evaluate the age and the trade limit and verify if the makers offer is inside that limit. The same in the other direction, the maker verifies with the takers local date and AgeWitness his age and limit. The exchanges local dates must not be diverging further than 1 day.
+
+
+== Gaming the scheme
+
+=== Broadcasting a back-dated AgeWitness object [[back-dating]]
+
+We need to be sure that the date of the trade in the AgeWitness object cannot be back-dated by a malicious trader. To achieve that, any node will reject AgeWitness objects which are older than 1 day. Additionally such a misbehaviour will trigger a ban on the network connection layer.
+
+
+=== Hijacking a foreign AgeWitness [[hijacking]]
+
+A more advanced fraud approach would be an attempt of hijacking someone else's AgeWitness and payment account to gain the benefit of an already aged account.
+
+A malicious trader could make a trade with someone who has already an old account and takes the account data of that trader (it is enough to take the min. set of uniquely identifying payment account data like IBAN + BIC) to use it for an own account. That fake account can only be used for buying BTC because for selling he would not receive the Fiat money but the user from where he has "stolen" the data. Because he has traded with the peer he has his salt as well so the verification of account data + salt would not detect that fraud. To protect against such an hijacking attempt we use the signed nonce to proof ownership of the AgeWitness data. Without the private key the fraudster cannot create a signature matching the public key in the AgeWitness.
+
+
+=== Self trade
+
+Any user could make a self trade with using a second application. This does not cause any risk because he proves that he is in possession of the payment account data (IBAN, BIC) and that is all we want to proof. If the trade was done with another user or not is not relevant here.
```

Comment:
don't understand this, suppose I do a self trade with myself and never do an actual fiat payment but just click the 'payment received' button. No actual payment has happened, the scammer can wait 2 months and transact the full amount. The only protection we have in such a case is the transaction limit + the fact that the scammer has to wait 2 months before doing something.

Reply:
The scheme assumes that the scammer does not want to wait 2 months when he has a stolen account. He does not know the IBAN/BIC before so he can only make a self-trade once he has the account. But to wait 2 months is a risk that it gets discovered in the meantime. Sure a real trade (bank transfer) gives even stronger protection. 
Will add a comment to make that more clear.

Reply:
New text:

=== Self trade

Any user could make a self trade with using a second application. This does not cause any risk because he proves that he is in possession of the payment account data (IBAN, BIC) and that is all we want to proof. If the trade was done with another user or not is not relevant here. Our basic assumption is that a fraudster does not want to wait a few months until he can use the stolen account for cashing out as he risks that the fraud get detected in the meantime.